starfish/core/image/_filter/clip_value_to_zero.py

```diff
@@ -0,0 +1,105 @@
+from functools import partial
+from typing import Optional, Union
+
+import numpy as np
+import xarray as xr
+
+from starfish.core.imagestack.imagestack import ImageStack
+from starfish.core.util import click
+from ._base import FilterAlgorithmBase
+from .util import determine_axes_to_group_by
+
+
+class ClipValueToZero(FilterAlgorithmBase):
+    """
+    Image clipping filter that clips values below a minimum value and above a
+    maximum value. The filter then subtracts the minimum value from the
+    clipped image.
+
+    By default, the min and max values are set to 0.0 and None respectively,
+    which will result in the filter doing nothing.
+
+    This is a wrapper for :py:func:`numpy.clip`.
+
+    Parameters
+    ----------
+    v_min : float, optional
+        Values below v_min are set to v_min. v_min is then subtracted from the
+        entire image (default 0)
+    v_max : int
```

Comment:
It should be this if your code is the source of truth.

```suggestion
    v_max : Optional[Number]
```

Reply:
`int` is definitely wrong but I think I'll stick with `Optional[float]` since `Number` is probably too broad (I don't think `np.clip` can handle `Complex` types?)

Also:

> [Rather than requiring that users write import numbers and then use numbers.Float etc., this PEP proposes a straightforward shortcut that is almost as effective: when an argument is annotated as having type float, an argument of type int is acceptable; similar, for an argument annotated as having type complex, arguments of type float or int are acceptable. This does not handle classes implementing the corresponding ABCs or the fractions.Fraction class, but we believe those use cases are exceedingly rare.](https://www.python.org/dev/peps/pep-0484/#the-numeric-tower)

Reply:
`Number` is a starfishism (starfish.core.types.Number) that means int or float.

Reply:
oh, thought you were talking about python `Number` class from the `numbers` module.